source/GameInterface/Services/MobileParties/Handlers/SettlementExitEnterHandler.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Common.Logging;
+using Common.Messaging;
+using Common.Network;
+using GameInterface.Services.MobileParties.Messages;
+using GameInterface.Services.ObjectManager;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using TaleWorlds.CampaignSystem.Actions;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Settlements;
+using TaleWorlds.CampaignSystem.ViewModelCollection.Map.MapBar;
+
+namespace GameInterface.Services.MobileParties.Handlers
+{
+    /// <summary>
+    /// Handles changes to parties for settlement entry and exit.
+    /// </summary>
+    public class SettlementExitEnterHandler : IHandler
+    {
+        private readonly IMessageBroker messageBroker;
+        private readonly IObjectManager objectManager;
+        private readonly ILogger Logger = LogManager.GetLogger<SettlementExitEnterHandler>();
+
+        public SettlementExitEnterHandler(IMessageBroker messageBroker, IObjectManager objectManager)
+        {
+            this.messageBroker = messageBroker;
+            this.objectManager = objectManager;
+            messageBroker.Subscribe<NetworkPartyEnteredSettlement>(Handle);
```

Comment:
The game interface should not know about the network, this should be handled in `Coop.Core.Client`
A new event will have to be created that this class will catch that is fired by the handler in `Coop.Core.Client`